main.go

```diff
@@ -311,7 +311,11 @@ func commandStatus(card *scard.Card) error {
 
 func commandShell(card *scard.Card) error {
 	fi, _ := os.Stdin.Stat()
-	if (fi.Mode() & os.ModeCharDevice) == 0 {
+	// Alex: I don't understand what the intention is here, but the condition
+	// breaks for me. I'm also guessing the error message doesn't totally
+	// capture the failing condition. 
+	// if (fi.Mode() & os.ModeCharDevice) == 0 {
+	if (fi.Mode() & os.ModeCharDevice) == os.ModeCharDevice {
```

Comment:
here we are checking the shell usage, if it's used pipe, redirect, or just the shell directly.
I would avoid using the shell directly because we should implement a full shell with history and allow moving the cursor.
But also because I would like to parse all the input before sending all the commands to the card, so that we can avoid sending "dangerous" commands if some conditions are not satisfied.
Foe this reason I would prefer using the shell command with pipe:
```
echo "keycard-select" | keycard shell
```

or redirect:

```
keycard shell <<END
keycard-select
keycard-...
...
END
```